website/api/controllers/entrance/signup.js

```diff
@@ -115,6 +131,43 @@ the account verification message.)`,
       });
     }
 
+    // If "Try Fleet Sandbox" was provided as the signupReason, this is a user signing up to try Fleet Sandbox.
+    if(signupReason === 'Try Fleet Sandbox') {
+
+      // Creating an expiration JS timestamp for the Fleet sandbox instance. NOTE: We send this value to the cloud provisioner API as an ISO 8601 string.
+      let fleetSandboxExpiresAt = Date.now() + (24*60*60*1000);
+
+      // Send a POST request to the cloud provisioner API
+      let cloudProvisionerResponse = await sails.helpers.http.post('https://sandbox.fleetdm.com/new', {
+        'name': firstName + ' ' + lastName,
+        'email': emailAddress,
+        'password': newUserRecord.password, //« Sending the hashed password to the Fleet Sandbox instance
+        'sandbox_expiration': new Date(fleetSandboxExpiresAt).toISOString(), // sending expiration_timestamp as an ISO string.
+      })
+      .timeout(5000)
+      .intercept('non200Response', 'couldNotProvisionSandbox');
+
+      if(cloudProvisionerResponse.URL) {
+        // Update the user's record with the fleetSandboxURL, fleetSandboxExpiresAt, and fleetSandboxKey.
+        await User.updateOne({id: newUserRecord.id}).set({
+          fleetSandboxURL: cloudProvisionerResponse.URL,
+          fleetSandboxExpiresAt: fleetSandboxExpiresAt,
+        });
+        // Start polling the /healthz endpoint of the created Fleet Sandbox instance, once it returns a 200 response, we'll continue.
+        await sails.helpers.flow.until( async()=>{
+          let serverResponse = await sails.helpers.http.sendHttpRequest('GET', cloudProvisionerResponse.URL+'/healthz')
+          .timeout(5000)
+          .tolerate('non200Response')
+          .tolerate('requestFailed');
+          if(serverResponse && serverResponse.statusCode) {
+            return serverResponse.statusCode === 200;
+          }
+        });
```

Comment:
@eashaw Please set up the other argument that limits how many times this can run, or how long (I forget, check getDef for how to use this)

Reply:
I added a 10-second timeout to `sails.helpers.flow.until()`